inspired by: mt8186-mt6359-add-sound-dai-cells-property.patch
from: https://patchwork.kernel.org/project/linux-mediatek/patch/20241210-genio700-audio-output-v2-1-c50886ae0be2@collabora.com/
see also: https://github.com/gregkh/linux/commit/9bc8353be720ca1f9cb6e03825929bc172e1157d

but: does not seem to help ...

diff --git a/arch/arm64/boot/dts/mediatek/mt6358.dtsi b/arch/arm64/boot/dts/mediatek/mt6358.dtsi
index e23672a2eea4..3d94302731c3 100644
--- a/arch/arm64/boot/dts/mediatek/mt6358.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6358.dtsi
@@ -9,6 +9,7 @@ pmic: pmic {
 		compatible = "mediatek,mt6358";
 		interrupt-controller;
 		#interrupt-cells = <2>;
+		#sound-dai-cells = <1>;
 
 		pmic_adc: adc {
 			compatible = "mediatek,mt6358-auxadc";
diff --git a/arch/arm64/boot/dts/mediatek/mt8186-corsola.dtsi b/arch/arm64/boot/dts/mediatek/mt8186-corsola.dtsi
index cfcc7909dfe6..6617fe4e96d8 100644
--- a/arch/arm64/boot/dts/mediatek/mt8186-corsola.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8186-corsola.dtsi
@@ -4,6 +4,7 @@
  */
 /dts-v1/;
 #include "mt8186.dtsi"
+#include "mt6358.dtsi"
 #include <dt-bindings/pinctrl/mt8186-pinfunc.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
